routing_model/src/scenario_tests.rs

```diff
@@ -825,6 +825,110 @@ mod dst_tests {
             &AssertState::default(),
         );
     }
+
+    #[test]
+    fn test_local_event_offline() {
```

Comment:
I would probably have a single test processing both NodeDetectedOffline and NodeDetectedOnline on different nodes, combining together test_local_event_offline & test_local_event_online_again since they test very similar things.